test/internet/test-dgram-multicast-multi-process.js

```diff
@@ -1,18 +1,115 @@
 'use strict';
-var common = require('../common'),
-    assert = require('assert'),
-    dgram = require('dgram'),
-    util = require('util'),
-    Buffer = require('buffer').Buffer,
-    fork = require('child_process').fork,
-    LOCAL_BROADCAST_HOST = '224.0.0.114',
-    TIMEOUT = common.platformTimeout(5000),
-    messages = [
-      new Buffer('First message to send'),
-      new Buffer('Second message to send'),
-      new Buffer('Third message to send'),
-      new Buffer('Fourth message to send')
-    ];
+const common = require('../common');
+const assert = require('assert');
+const dgram = require('dgram');
+const Buffer = require('buffer').Buffer;
+const fork = require('child_process').fork;
+const LOCAL_BROADCAST_HOST = '224.0.0.114';
+const TIMEOUT = common.platformTimeout(5000);
+const messages = [
+        new Buffer('First message to send'),
+        new Buffer('Second message to send'),
+        new Buffer('Third message to send'),
+        new Buffer('Fourth message to send')
+      ];
+
+// skip test in FreeBSD jails
+if (common.inFreeBSDJail) {
+  console.log('1..0 # Skipped: In a FreeBSD jail');
+  return;
+}
+
+function launchChildProcess(index) {
+  const worker = fork(process.argv[1], ['child']);
+  workers[worker.pid] = worker;
+
+  worker.messagesReceived = [];
+
+  //handle the death of workers
+  worker.on('exit', function(code, signal) {
+    // don't consider this the true death if the
+    // worker has finished successfully
+
+    // or if the exit code is 0
+    if (worker.isDone || code === 0) {
+      return;
+    }
+
+    dead += 1;
+    console.error('[PARENT] Worker %d died. %d dead of %d',
+                  worker.pid,
+                  dead,
+                  listeners);
+
+    if (dead === listeners) {
+      console.error('[PARENT] All workers have died.');
+      console.error('[PARENT] Fail');
+
+      killChildren(workers);
+
+      process.exit(1);
+    }
+  });
+
+  worker.on('message', function(msg) {
+    if (msg.listening) {
+      listening += 1;
+
+      if (listening === listeners) {
+        //all child process are listening, so start sending
+        sendSocket.sendNext();
+      }
+    }
+    else if (msg.message) {
```

Comment:
I'll do a `return` at the end of the previous if block and get rid of the `else` altogether.